### PR TITLE
Add basic HttpRequestor Gem documentation

### PR DESCRIPTION
**Pull request by lmbr-pip**
Quick port and update of Lumberyard 1.x documentation
- Added working sample and updated syntax

Signed-off-by: Pip Potter <rppotter@amazon.com>

### CODE REVIEW

File: content/docs/user-guide/gems/reference/network/http-requestor.md
Comment (sptramer) [inline content/docs/user-guide/gems/reference/network/http-requestor.md:12]:
```suggestion
{{ note}} 
This feature is currently supported only on Windows.
{{ /note }}
```

File: content/docs/user-guide/gems/reference/network/http-requestor.md
Comment (sptramer) [inline content/docs/user-guide/gems/reference/network/http-requestor.md:25]:
````suggestion
The `HttpRequestor` Gem uses the AWS C++ SDK. To pick up these dependencies, add the following to your project  CMake file:

```
BUILD_DEPENDENCIES
        PRIVATE
            ...
            Gem::HttpRequestor
            3rdParty::AWSNativeSDK::Core
```
````

File: content/docs/user-guide/gems/reference/network/http-requestor.md
Comment (sptramer) [inline content/docs/user-guide/gems/reference/network/http-requestor.md:27]: This content will be accepted for now, but should be placed in the header files of the `HttpRequestor` Gem so that we can move from hand-authored API reference on the docs to API reference generated from the headers themselves:

https://o3de.org/docs/api/gems/httprequestor/class_http_requestor_1_1_http_requestor_requests.html

File: content/docs/user-guide/gems/reference/network/http-requestor.md
Comment (sptramer) [inline content/docs/user-guide/gems/reference/network/http-requestor.md:29]:
```suggestion
The HttpRequestor Gem has separate sets of APIs for adding requests and adding text requests.
```

File: content/docs/user-guide/gems/reference/network/http-requestor.md
Comment (sptramer) [inline content/docs/user-guide/gems/reference/network/http-requestor.md:15]:
```suggestion
To use the HttpRequestor Gem, you must enable it in your project. For more information refer to [Adding and Removing Gems in a Project](/docs/user-guide/project-config/add-remove-gems/).
```

File: content/docs/user-guide/gems/reference/network/http-requestor.md
Comment (sptramer) [inline content/docs/user-guide/gems/reference/network/http-requestor.md:33]:
```suggestion
Use the `AddRequest`, `AddRequestWithHeaders`, and `AddRequestWithHeadersAndBody` APIs to send generic HTTP requests to any website and receive the returned data in JSON format. The methods return the data received in the `callback` parameter.
```

File: content/docs/user-guide/gems/reference/network/http-requestor.md
Comment (sptramer) [inline content/docs/user-guide/gems/reference/network/http-requestor.md:62]:
```suggestion
| URI | `AZStd::String` | The fully qualified web address, in the following format: `scheme:[//[user:password@]host[:port]][/]path[?query][#fragment]` | 
| method | `Aws::Http::HttpMethod` | The method type. Supports GET, POST, DELETE, PUT, HEAD, and PATCH HTTP methods. | 
| callback |  | This function is called when the HTTP request is completed. The response body and code are present in the callback. | 
| headers | `HttpRequestor::Headers` | The list of header fields for the HTTP request. | 
| body | `AZStd::String` | Optional body to send with the request. | 
```

**NOTE:** The type for `callback` is missing. Function pointer types are extra important to annotate.

File: content/docs/user-guide/gems/reference/network/http-requestor.md
Comment (sptramer) [inline content/docs/user-guide/gems/reference/network/http-requestor.md:54]:
```suggestion
```

File: content/docs/user-guide/gems/reference/network/http-requestor.md
Comment (sptramer) [inline content/docs/user-guide/gems/reference/network/http-requestor.md:64]:
```suggestion
#### Returns

`void`
```

File: content/docs/user-guide/gems/reference/network/http-requestor.md
Comment (sptramer) [inline content/docs/user-guide/gems/reference/network/http-requestor.md:77]:
```suggestion
```

File: content/docs/user-guide/gems/reference/network/http-requestor.md
Comment (sptramer) [inline content/docs/user-guide/gems/reference/network/http-requestor.md:82]:
```suggestion
| json | `Aws::Utils::Json::JsonValue` | The JSON object. The life span of this object is valid only during the scope of the callback. | 
| responseCode | `Aws::Http::HttpResponseCode` | The [HTTP response code](https://datatracker.ietf.org/doc/html/rfc7231#section-6). | 
```

Comment (sptramer) [reply content/docs/user-guide/gems/reference/network/http-requestor.md:82]: Feel free to reject adding the link to the IETF doc, but in my experience it's useful to remind programmers of _extensive_ external reference where useful.

File: content/docs/user-guide/gems/reference/network/http-requestor.md
Comment (sptramer) [inline content/docs/user-guide/gems/reference/network/http-requestor.md:88]:
```suggestion
Use the `AddTextRequest`, `AddTextRequestWithHeaders`, and `AddTextRequestWithHeadersAndBody` APIs to send a generic HTTP request to any website and receive the returned data in a text string. The methods return the data received in the `callback` parameter.
```

File: content/docs/user-guide/gems/reference/network/http-requestor.md
Comment (sptramer) [inline content/docs/user-guide/gems/reference/network/http-requestor.md:119]:
````suggestion
| URI | `AZStd::String` | The fully qualified web address, in the following format: `scheme:[//[user:password@]host[:port]][/]path[?query][#fragment]` | 
| method | `Aws::Http::HttpMethod` | The method type. Supports GET, POST, DELETE, PUT, HEAD, and PATCH HTTP methods. | 
| callback |  | This function is called when the HTTP request is completed. The response body and code are present in the callback. | 
| headers | `HttpRequestor::Headers` | The list of header fields for the HTTP request. | 
| body | `AZStd::String` | Optional body to send with the request. | 

#### Returns

`void`
````

File: content/docs/user-guide/gems/reference/network/http-requestor.md
Comment (sptramer) [inline content/docs/user-guide/gems/reference/network/http-requestor.md:132]:
```suggestion
```

File: content/docs/user-guide/gems/reference/network/http-requestor.md
Comment (sptramer) [inline content/docs/user-guide/gems/reference/network/http-requestor.md:137]:
```suggestion
| response | `AZStd::string&` | The text returned from the server. The life span of this object is valid only during the scope of the callback. | 
| responseCode | `Aws::Http::HttpResponseCode` | The [HTTP response code](https://datatracker.ietf.org/doc/html/rfc7231#section-6). | 
```

File: content/docs/user-guide/gems/reference/network/http-requestor.md
Comment (sptramer) [inline content/docs/user-guide/gems/reference/network/http-requestor.md:139]:
```suggestion
#### Returns

`void`
```

File: content/docs/user-guide/gems/reference/network/http-requestor.md
Comment (sptramer) [inline content/docs/user-guide/gems/reference/network/http-requestor.md:32]: 👍 👍 👍 👍 👍 👍 👍

**Review by sptramer — APPROVED** [review]

**Comment by sptramer** [comment]
@junbo75 - Will merge after your approval.

File: content/docs/user-guide/gems/reference/network/http-requestor.md
Comment (junbo75) [inline content/docs/user-guide/gems/reference/network/http-requestor.md:131]: Minor: For the type of response, should it be AZStd::string& or AZStd::string?

**Review by junbo75 — APPROVED** [review]
LG. Thanks for adding this document.

**Pull request merged**
